### PR TITLE
Add ASP.NET distributed tracing example

### PR DESCRIPTION
**Pull request by noahfalk**
Adds an ASP.NET example for OpenTelemetry distributed tracing

Fixes https://github.com/dotnet/diagnostics/issues/3451

### CODE REVIEW

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 1]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
 
 ## Collect traces using OpenTelemetry
 
-### Prerequisites
+[OpenTelemetry](https://opentelemetry.io/) is a vendor-neutral open-source project supported by the
+[Cloud Native Computing Foundation](https://www.cncf.io/) that aims to standardize generating and collecting telemetry for
+cloud-native software. In these examples, you will collect and display distributed trace information on the console though
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:22]:
```suggestion
cloud-native software. In these examples, you'll collect and display distributed trace information on the console, though
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 2]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+OpenTelemetry can be configured to send it elsewhere. For more information, see the
+[OpenTelemetry getting started guide](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/getting-started/README.md).
+
+### ASP.NET Example
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:26]:
```suggestion
### ASP.NET example
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 3]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+
+#### Prerequisites
+
+- [.NET Core 7.0 SDK](https://dotnet.microsoft.com/download/dotnet) or a later version
+
+#### Create an example application
+
+First, create a new ASP.NET webapp to use as the demo application
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:34]:
```suggestion
First, create a new ASP.NET web app to use as the demo application.
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 4]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+
+```dotnetcli
+dotnet new webapp
+```
+
+This app can now run and display a web page, but no distributed tracing information is being collected yet
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:40]:
```suggestion
This app displays a web page, but no distributed tracing information is collected yet
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 5]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+if we browse the web page.
+
+#### Configuring collection
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:43]:
```suggestion
#### Configure collection
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 6]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+
+To use OpenTelemetry we need to add references to several NuGet packages.
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:45]:
```suggestion
To use OpenTelemetry, you need to add references to several NuGet packages.
```

**Comment by noahfalk** [comment]
@mikelle-rogers - could you take a look please?

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 7]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+dotnet add package OpenTelemetry.Exporter.Console --version 1.4.0-rc1
+dotnet add package OpenTelemetry.Extensions.Hosting --version 1.4.0-rc1
+dotnet add package OpenTelemetry.Instrumentation.AspNetCore --version 1.0.0-rc9.10
+```
+
+> [!NOTE]
+> At the time of writing, the 1.4.0 Release Candidate 1 build was the latest version of OpenTelemetry available. Once a final version
+> is available, use that instead.
+
+Next modify the source code of Program.cs so it looks like this:
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:58]:
```suggestion
Next, modify the source code in *Program.cs* so it looks like this:
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 8]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+app.UseRouting();
+
+app.UseAuthorization();
+
+app.MapRazorPages();
+
+app.Run();
+```
+
+Run the app and use a web browser to browse to the web page being hosted. Now that we enabled OpenTelemetry distributed
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:97]:
```suggestion
Run the app and use a web browser to browse to the web page being hosted. Now that you enabled OpenTelemetry distributed
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 9]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+tracing you should see information about the browser web requests printed to the console:
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:98]:
```suggestion
tracing, you should see information about the browser web requests printed to the console:
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 10]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+    http.target: /
+    http.url: http://localhost:5163/
+    http.flavor: 1.1
+    http.user_agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36 Edg/108.0.1462.76
+    http.status_code: 200
+Resource associated with Activity:
+    service.name: unknown_service:demo
+```
+
+All of the OpenTelemetry configuration is occuring in the new source lines that start with `builder.Services.AddOpenTelemetry()`. We used
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:123]:
```suggestion
All of the OpenTelemetry configuration occurs in the new source lines that start with `builder.Services.AddOpenTelemetry()`. You used
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 11]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+`.WithTracing(...)` to enable distributed tracing. `AddAspNetCoreInstrumentation()` enabled OpenTelemetry to collect all the
+distributed trace Activities that are produced by the ASP.NET Core web server and `AddConsoleExporter()` instructs OpenTelemetry
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:125]:
```suggestion
distributed trace Activities that are produced by the ASP.NET Core web server, and `AddConsoleExporter()` instructs OpenTelemetry
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 12]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+to send that information to the console. For a less trivial app you might want to add more instrumentation libraries to also collect
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:126]:
```suggestion
to send that information to the console. For a less trivial app, you might add more instrumentation libraries to also collect
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 13]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+tracing for database queries or out-bound http requests. You would also replace the console exporter with an exporter for Jaeger, Zipken, or
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:127]:
```suggestion
tracing for database queries or outbound HTTP requests. You'd also replace the console exporter with an exporter for Jaeger, Zipken, or
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 14]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+some other monitoring service you have chosen to use.
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:128]:
```suggestion
another monitoring service you've chosen to use.
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 15]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+
+### Console App Example
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:130]:
```suggestion
### Console app example
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 16]
```diff
@@ -17,11 +17,123 @@ available to diagnose application issues when needed. See
...
+
+#### Prerequisites
 
 - [.NET Core 2.1 SDK](https://dotnet.microsoft.com/download/dotnet) or a later version
 
-### Create an example application
+#### Create an example application
 
 Before any distributed trace telemetry can be collected, we need to produce it. Often this instrumentation might be
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:138]:
```suggestion
Before any distributed trace telemetry can be collected, you need to produce it. Often this instrumentation is
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md [hunk 17]
```diff
@@ -95,13 +207,7 @@ Running the app does not collect any trace data yet:
...
 ```
 
-### Collect using OpenTelemetry
-
-[OpenTelemetry](https://opentelemetry.io/) is a vendor-neutral open-source project supported by the
-[Cloud Native Computing Foundation](https://www.cncf.io/) that aims to standardize generating and collecting telemetry for
-cloud-native software. In this example, you will collect and display distributed trace information on the console though
-OpenTelemetry can be reconfigured to send it elsewhere. For more information, see the
-[OpenTelemetry getting started guide](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/getting-started/README.md).
+#### Configuring collection
```
Comment (gewarren) [inline docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:210]:
```suggestion
#### Configure collection
```

**Review by gewarren — APPROVED** [review]

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md (hunk 11 above)
Comment (mikelle-rogers) [reply docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:125]:
```suggestion
distributed trace Activities that are produced by the ASP.NET Core web server. `AddConsoleExporter()` instructs OpenTelemetry
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md (hunk 12 above)
Comment (mikelle-rogers) [reply docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:126]:
```suggestion
to send that information to the console. For a less trivial app, you could add more instrumentation libraries to also collect
```

File: docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md (hunk 13 above)
Comment (mikelle-rogers) [reply docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md:127]:
```suggestion
tracing for database queries or outbound HTTP requests. You could also replace the console exporter with an exporter for Jaeger, Zipken, or
```

**Review by mikelle-rogers — COMMENTED** [review]
Looks great as far as I can tell!

**Comment by noahfalk** [comment]
Thanks @mikelle-rogers and @gewarren :)

**Pull request merged**